### DISCUSSION TITLE
Add useUpsert operation

### DISCUSSION THREAD

**Discussion by kachar** [tmm/react-supabase#5] [Ideas]
While supabase gives devs the opportunity to do insert or update with a single command, this feature doesn't seem supported by `react-supabase` lib

https://supabase.io/docs/reference/javascript/upsert

```ts
const { data, error } = await supabase.from('messages').upsert({
  id: 3,
  message: 'foo',
  username: 'supabot'
})
```

My suggestion is to include it as `useUpsert` hook

```ts
const [{ count, data, error, fetching }, execute] = useUpsert('messages')

async function onClickUpsert(id, message, username) {
  const { count, data, error } = await execute({
    id,
    message,
    username,
  })
}
```

**Comment by tmm** [comment]
Great idea. It should be easy to add. Feel free to open a pull request or I'll add to my list.

**Reply by kachar** [reply]
@tmm Added a PR for the given functionality at https://github.com/tmm/react-supabase/pull/6
